File: 100-999/101. Symmetric Tree/solution_1.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution
{
    public void WalkTree(TreeNode left, TreeNode right, ref bool symmetric)
    {
        if (left != null && right != null)
        {
            if (left.val != right.val)
            {
                symmetric = false;
                return;
            }

            WalkTree(left.left, right.right, ref symmetric);
            WalkTree(left.right, right.left, ref symmetric);
        }
        else if (left != null && right == null || left == null && right != null)
        {
            symmetric = false;
            return;
        }
    }
    public bool IsSymmetric(TreeNode root)
    {
        bool symmetric = true;

        if (root.left != null && root.right != null)
            WalkTree(root.left, root.right, ref symmetric);
        else if (root.left != null && root.right == null || root.left == null && root.right != null)
            symmetric = false;

        return symmetric;
    }
}

File: 100-999/101. Symmetric Tree/solution_1.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */

/**
 * @param {TreeNode} left
 * @param {TreeNode} right
 * @param {object} symmetric
 * @return {boolean}
 */

var walkTree = function(left, right, symmetric)
{
    if (left && right)
    {
        if (left.val != right.val)
        {
            symmetric.value = false
            return
        }

        walkTree(left.left, right.right, symmetric)
        walkTree(left.right, right.left, symmetric)
    }
    else if (left && !right || !left && right)
    {
        symmetric.value = false
        return
    }
}

/**
 * @param {TreeNode} root
 * @return {boolean}
 */

var isSymmetric = function(root) 
{
    const symmetric = {value: true}
    
    if (root.left && root.right)
    {
        walkTree(root.left, root.right, symmetric)
    }
    else if (root.left && !root.right || !root.left && root.right)
        symmetric.value = false

    return symmetric.value
};